File: basic_python/HELLOPYTHON/day08/speedinsert.py
import cx_Oracle # cx: c++ 가져다 쓰는거라서 속도가 개선됨 
import time
# import datetime
# startd = datetime.datetime().now()
# endd = datetime.datetime().now()

connection = cx_Oracle.connect('python','python','localhost:1521/xe')
cursor = connection.cursor()
sql_str = "insert into sample(col01, col02, col03) values(:1,:2,:3)"

start = time.time() #시작시간
for i in range(10000):
    cursor.execute(sql_str,('77','77','77')) # 행운의 77 77 77
end = time.time() #끝시간

connection.commit()
cursor.close()
connection.close()

total = end-start
print(total)

File: basic_python/HELLOJAVA/src/day07/SpeedInsert.java
package day07;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.Statement;
import java.util.Calendar;

public class SpeedInsert {
	
	public static void main(String[] args) throws Exception {
		

		String driver = "oracle.jdbc.driver.OracleDriver";
		String url = "jdbc:oracle:thin:@localhost:1521:xe"; 
		
		Class.forName(driver);
		Connection conn = DriverManager.getConnection(url, "python", "python");
		Statement stmt = conn.createStatement();
		
		long now1 = Calendar.getInstance().getTimeInMillis();
		
		for(int i = 0;i<10000;i++) {
			String sql = "insert into sample(col01, col02, col03) values ('4','4','4')";
			int cnt = stmt.executeUpdate(sql);
		}
		
		long now2 = Calendar.getInstance().getTimeInMillis();
				
		stmt.close();
		conn.close();
		

		
		long ellapse = now2 - now1;
		
		System.out.println("ellapse: "+ellapse);
		
		
		
		
		

	}
	
	
}
